text/src/autogluon/text/automm/predictor.py

```diff
@@ -433,67 +421,130 @@ def fit(
+            teacher_predictor=None if hyperparameter_tune_kwargs is not None else teacher_predictor,
+            hpo_mode=(hyperparameter_tune_kwargs is not None)  # skip average checkpoint if in hpo mode
         )
+        
+        if hyperparameter_tune_kwargs is not None:
+            # TODO: allow custom gpu
+            resources = dict(num_gpus=torch.cuda.device_count())
+            if _fit_args['max_time'] is not None:
+                _fit_args['max_time'] *= 0.95  # give some buffer time to ray lightning trainer
+            predictor = self._hyperparameter_tune(
+                hyperparameter_tune_kwargs=hyperparameter_tune_kwargs,
+                resources=resources,
+                **_fit_args
+            )
+            return predictor
+        
+        self._fit(**_fit_args)
         return self
-
+    
+    def _hyperparameter_tune(self, hyperparameter_tune_kwargs, resources, **_fit_args):
+        from autogluon.core.hpo import run, cleanup_trials, EmptySearchSpace, AutommRayTuneAdapter
+        search_space = _fit_args.get('hyperparameters', dict())
+        metric = 'val_' + _fit_args.get('validation_metric_name')
+        mode = _fit_args.get('minmax_mode')
+        save_path = _fit_args.get('save_path')
+        time_budget_s = _fit_args.get('max_time')
+        try:
+            analysis = run(
+                trainable=self._hpo_fit_wrapper,
+                trainable_args=_fit_args,
+                search_space=search_space,
+                hyperparameter_tune_kwargs=hyperparameter_tune_kwargs,
+                metric=metric,
+                mode=mode,
+                save_dir=save_path,
+                ray_tune_adapter=AutommRayTuneAdapter(),
+                total_resources=resources,
+                time_budget_s=time_budget_s,
+                keep_checkpoints_num=3,  # TODO: find a way to extract this from config. Might need to separate generate config and trial specific config
+                checkpoint_score_attr=metric,
+                verbose=2
+            )
+        except EmptySearchSpace:
+            raise ValueError("Please provide a search space in order to do hyperparameter tune")
+        except Exception as e:
+            raise e
+        else:
+            # find the best trial
+            best_trial = analysis.get_best_trial(
+                metric=metric,
+                mode=mode,
+            )
+            if best_trial is None:
+                raise ValueError("AutoMMPredictor wasn't able to find the best trial. It's likely that the time is not enough to train a single epoch for trials.")
+            # clean up other trials
+            logger.info('Removing non-optimal trials and only keep the best one.')
+            cleanup_trials(save_path, best_trial.trial_id)
+            best_trial_path = os.path.join(save_path, best_trial.trial_id)
+            # reload the predictor metadata
+            predictor = self.__class__._load_metadata(path=best_trial_path)
```

Comment:
I think there is a way to avoid creating a new predictor. We can let a predictor object as the input of `_load_metadata` and pass `self` into `_load_metadata`.